crates/holofuel_init/src/reserve_init.rs

```diff
@@ -1,14 +1,22 @@
 use anyhow::Result;
+use holochain_types::dna::hash_type::Agent;
+use holochain_types::dna::HoloHash;
 use holochain_types::prelude::{ExternIO, FunctionName, ZomeName};
 use hpos_hc_connect::holofuel_types::{Reserve, ReserveSalePrice, ReserveSetting};
 use hpos_hc_connect::HolofuelAgent;
-use tracing::log::trace;
-use tracing::{info, log::warn};
+use tracing::{info, instrument, trace, warn};
 
-pub async fn set_up_reserve(mut agent: HolofuelAgent) -> Result<()> {
+#[instrument(err, skip(agent))]
+pub async fn set_up_reserve(
+    mut agent: HolofuelAgent,
+    agent_pub_key: HoloHash<Agent>,
+) -> Result<()> {
     trace!("Setting up reserve settings...");
     match ReserveSetting::load_happ_file() {
-        Ok(reserve_settings) => {
+        Ok(mut reserve_settings) => {
+            let agent_pub_key_byte_arr: [u8; 32] =
+                <[u8; 32]>::try_from(agent_pub_key.get_raw_32())?;
+            reserve_settings.external_signing_key = agent_pub_key_byte_arr.into();
```

Comment:
This would still expect you to pass a dummy `external_signing_key` in the JSON file, do you want it that way?